be/src/storage/task/engine_batch_load_task.cpp

```diff
@@ -209,8 +196,19 @@ Status EngineBatchLoadTask::_push(const TPushReq& request, std::vector<TTabletIn
 
     if (request.tablet_type == TTabletType::TABLET_TYPE_LAKE) {
         auto tablet_id = request.tablet_id;
-        auto tablet_version = (request.version == -1) ? choose_any_version(tablet_id) : request.version;
-        //                    ^^^^^^^^^^^^^^^^^^^^^^^ This should only happen when the FE version is less than 3.3.0
+        // Starting from 3.3.0, the tablet schema in different versions of the tablet metadata may be different. We
+        // need to read the tablet of a specific version (request.version) to ensure that the tablet schema used
+        // during import meets the expectations. However, if the FE version is lower than 3.3.0, the request.version
+        // will always be set to -1 by the FE. At this time, we can read any version of the tablet metadata, because
+        // if the FE version is lower than 3.3.0, it means that there will be no fast schema evolution, and the tablet
+        // schema in all versions of the tablet metadata will be the same.
+        auto tablet_version = request.version;
+        if (tablet_version == -1) {
```

Comment:
it takes a strong assumption that FE will always set `-1` as indicating no version specified, can we assume that all `<=0` is a VERSION_NOT_SET?

Reply:
If the version is not -1, but 0 or other negative values, then this unexpected behavior is not simply caused by the FE version less than 3.3.0. At this time, perhaps we should let BE return an error instead of going to the choose_any_version branch